### ISSUE TITLE
Comment Code to Divide Code into MVC

### ISSUE THREAD

**Issue by gbusto** [gbusto/Ballz1#148]
I think there would be a lot of benefit to changing to an MVC architecture. This is because I'm at the point where I want to save game state, but the lines between objects to save aren't totally clean. And it will get more difficult as we add in upgrades that the user has, how much currency they have, etc.

**Model**: All rules of the game world are implemented in the model. The model also contains the state of every game object. Any object that a model contains need to conform to the NSCoding protocol so we can serialize it and save it to disk. When we add different game modes (for example, levels, continuous play, etc), each game mode will have a model.

**View**: The view uses the model to know where to draw things. The model and controller don't know anything about the screen layout or how to display objects. The view will be the one that calls `scene.addChild(SKNode)`. When we add themes (for holidays) or color schemes all we'll have to do is change the view code to display things differently. Also, the code for handling different devices (iPhone 7/8 vs iPhone X+ vs iPad) can be separated out into the view.

**Controller**: Handles user input and manipulates the model.

**tldr;** **model** contains game rules and object states and is manipulated by the **controller**, the **view** renders the model to the screen.

The problem is I'm not sure how to split everything out. So in this issue, I'll go through and comment all functions and control blocks to say Model, View, or Controller.
